mirdata/datasets/haydn_op20.py

```diff
@@ -0,0 +1,310 @@
+)
+
+
+class Track(core.Track):
+    """haydn op20 track class
+
+    Args:
+        track_id (str): track id of the track
+
+    Attributes:
+        title (str): title of the track
+        track_id (str): track id
+        humdrum_annotated_path (str): path to humdrum annotated score
+
+    Cached Properties:
+        key (str): key annotation
+        score (np.array): computed audio spectrum
+        keys (list): annotated local key
+        roman_numerals (list): MusicBrainz metadata
+        chords (list): annotated local key
+        duration (int): relative duration
+        midi_path (str): path to midi
+        score (music21.stream.Score): music21 score
+    """
+
+    def __init__(
+        self,
+        track_id,
+        data_home,
+        dataset_name,
+        index,
+        metadata,
+    ):
+        super().__init__(
+            track_id,
+            data_home,
+            dataset_name,
+            index,
+            metadata,
+        )
+        self.humdrum_annotated_path = os.path.join(
+            self._data_home, self._track_paths["annotations"][0]
+        )
+        self.title = os.path.splitext(self._track_paths["annotations"][0])[0]
+
+    @core.cached_property
+    def score(self) -> music21.stream.Score:
+        return load_score(self.humdrum_annotated_path)
+
+    @core.cached_property
+    def keys(self, resolution=28) -> List[dict]:
+        return load_key(self.humdrum_annotated_path, resolution)
+
+    @core.cached_property
+    def roman_numerals(self, resolution=28) -> List[dict]:
+        return load_roman_numerals(self.humdrum_annotated_path, resolution)
+
+    @core.cached_property
+    def chords(self, resolution=28) -> List[dict]:
+        return load_roman_numerals(self.humdrum_annotated_path, resolution)
+
+    @core.cached_property
+    def duration(self) -> int:
+        return self.chords[-1]["time"]
+
+    @core.cached_property
+    def midi_path(self) -> str:
+        return load_midi_path(self.humdrum_annotated_path)
+
+    def to_jams(self):
+        """Get the track's data in jams format
+
+        Returns:
+            jams.JAMS: the track's data in jams format
+
+        """
+        return jams_utils.jams_converter(
+            metadata={
+                "duration": self.duration,
+                "title": self.title,
+                "key": self.keys,
+                "chords": self.chords,
+                "roman_numerals": self.roman_numerals,
+                "midi_path": self.midi_path,
+                "humdrum_annotated_path": self.humdrum_annotated_path,
+            },
+        )
+
+
+def load_score(path):
+    """Load haydn op20 score with annotations from a file with music21 format (music21.stream.Score).
+
+    Args:
+        path: path to hrm annotations
+
+    Returns:
+        music21.stream.Score: score in music21 format
+
+    """
+    if not os.path.exists(path):
+        raise IOError
```

Comment:
This is not needed since the error that would be raised is `IOError` anyway and it will contain a useful message. Remove from following loaders as well